randovania/games/cave_story/exporter/patch_data_factory.py

```diff
@@ -42,12 +45,50 @@ def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.CAVE_STORY
 
     def create_game_specific_data(self) -> dict:
-        game_description = self.game
-        seed_number = self.description.get_seed_for_player(self.players_config.player_index)
-        music_rng = Random(seed_number)
-        mychar_rng = Random(seed_number)
-        hint_rng = Random(seed_number)
+        self.seed_number = self.description.get_seed_for_player(self.players_config.player_index)
+
+        self.maps = self._create_maps_data()
+        self.maps["Start"]["pickups"]["0201"] = self._create_starting_script()
+
+        data: CaverData = {
+            "maps": self.maps,
+            "other_tsc": {"Head": self._head_tsc_edits()},
+            "mychar": self.cosmetic_patches.mychar.mychar_bmp(Random(self.seed_number)),
+            "hash": get_ingame_hash(self.description.shareable_hash_bytes),
+            "uuid": f"{{{self.players_config.get_own_uuid()}}}",
+        }
+        return typing.cast(dict, data)
```

Comment:
is CaverData not inheriting from dict? is an explicit cast really necessary here?

Reply:
it is necessary, because dict type parameters are invariant. removing the cast causes an error